src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -61,18 +64,25 @@ public PersonCard(Person person, int displayedIndex) {
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
 
-        if (person.getSchedule().toString().isEmpty()) {
-            schedule.setText(person.getSchedule().toString());
-        } else {
-            schedule.setText(
-                    LocalDateTime.parse(person.getSchedule().toString(), DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"))
-                            .format(DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a")));
-        }
-        if (person.getSchedule().getNotes() == null || person.getSchedule().getNotes().isEmpty()) {
-            note.setText("");
+        // Handle multiple schedules
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
+
+        Set<Schedule> schedules = person.getSchedules();
+        if (schedules.isEmpty()) {
+            schedule.setText("No scheduled appointments");
         } else {
-            note.setText("Notes: " + person.getSchedule().getNotes());
+            String formattedSchedules = schedules.stream()
```

Comment:
Perhaps the formatting logic could be extracted to a separate method for better code readability?